packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -46,13 +46,13 @@ const INFINITE_REDIRECTION_LOOP_COOKIE = '__clerk_redirection_loop';
  * The default ideal matcher that excludes the _next directory (internals) and all static files,
  * but it will match the root route (/) and any routes that start with /api or /trpc.
  */
-export const DEFAULT_CONFIG_MATCHER = ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'];
+export const DEFAULT_CONFIG_MATCHER = ['/((?!.+[\\w]+\\.[\\w]+$|_next).*)', '/', '/(api|trpc)(.*)'];
```

Comment:
☁️ i think we should use the `.+\.[\w]+$` instead of `+[\w]+\\.[\w]+$` because filenames may contain symbols outside of `[a-zA-Z_0-9]` (eg `-`).

The next part is outside of the scope of this PR but I would like to open a discussion on that:
Could we split this regex into parts to make it more maintainable? Eg
```typescript
const allowAPIs = '/(api|trpc)(.*)';
const allowRootPaths = '/';
const ignoreNextUrls = '/(?!_next.*)';
const ignoreFileExtensions = '/(?!.+\.[\w]+$)';
export const DEFAULT_CONFIG_MATCHER = [allowAPIs, allowRootPaths, ignoreNextUrls, ignoreFileExtensions];
```

Reply:
Good suggestion. The negative rules need to remain together it seems like, otherwise one of them will match the other and they effectively cancel each other out, but I do like this!

Reply:
i would insist a bit on allowing any character before the file extension and not only the specified ones:
```
const VALID_FILENAME_CHARS = '\\w-\\$\\(\\)';
```
If the rest of the team agrees to keep only the limited set of chars, then you may proceed with merging it.

cc: @nikosdouvlis , @anagstef

Reply:
I agree with @dimkl! I also think we should allow any character (including `~`) in the file name.

Reply:
Sounds good 👍